test/ResponseTest.php

```diff
@@ -66,6 +66,122 @@ public function testReasonPhraseDefaultsToStandards()
+        $httpStatus = 0;
+
+        try {
+            $options = [
+                'http' => [
+                    'method'  => 'GET',
+                    'timeout' => 30,
+                ],
+            ];
+
+            $contents = file_get_contents(
+                'https://www.iana.org/assignments/http-status-codes/http-status-codes.xml',
+                false,
+                stream_context_create($options)
+            );
+
+            if ($http_response_header) {
+                if (preg_match('/^HTTP\/[0-9\.]+\s*([0-9]+)\s*.+$/i', $http_response_header[0], $matches) > 0) {
+                    $httpStatus = $matches[1];
+                }
+            }
+
+            if ($httpStatus == 200) {
+                $ianaHttpStatusCodes->loadXml($contents);
+                $validXml = $ianaHttpStatusCodes->relaxNGValidate(__DIR__ . '/TestAsset/http-status-codes.rng');
+
+                if ($validXml) {
+                    file_put_contents(__DIR__ . '/TestAsset/http-status-codes.xml', $contents);
+                    print 'IANA "http-status-codes.xml" updated successful' . "\n";
+                }
+            }
+        } catch (\Exception $e) {
+            $errorMessage = $e->getMessage();
+        }
+
+
+        if (! $validXml) {
+            if ($errorMessage) {
+                print 'Error on IANA "http-status-codes.xml" update. Error: ' . $errorMessage . "\n";
+            }
+
+            try {
+                $ianaHttpStatusCodes->load(__DIR__ . '/TestAsset/http-status-codes.xml');
+                $validXml = $ianaHttpStatusCodes->relaxNGValidate(__DIR__ . '/TestAsset/http-status-codes.rng');
+            } catch (\Exception $e) {
+                $errorMessage = $e->getMessage();
+            }
+        }
+
+        restore_error_handler();
+
+        if (! $validXml) {
+            $this->markTestIncomplete(
+                'Invalid IANA "http-status-codes.xml". Error: ' . $errorMessage
+            );
+            $ianaHttpStatusCodes = null;
+        }
+
+        return $ianaHttpStatusCodes;
+    }
+
+    public function ianaCodesReasonPhrasesProvider()
+    {
+        $ianaHttpStatusCodes = $this->updateAndLoadIanaHttpStatusCodes();
+
+        if (! $ianaHttpStatusCodes) {
+            return null;
+        }
+
+        $ianaCodesReasonPhrases = [];
+
+        $xpath = new \DomXPath($ianaHttpStatusCodes);
+        $xpath->registerNamespace('ns', 'http://www.iana.org/assignments');
+
+        $records = $xpath->query('//ns:record');
+
+        foreach ($records as $record) {
+            $value = $xpath->query('.//ns:value', $record)->item(0)->nodeValue;
+            $description = $xpath->query('.//ns:description', $record)->item(0)->nodeValue;
+
+            if ($description === 'Unassigned' || $description === '(Unused)') {
+                continue;
+            }
+
+            $range = preg_match('/^([0-9]+)\s*\-\s*([0-9]+)$/', $value, $matches);
+
+            if (! $range) {
```

Comment:
Swap the conditional to avoid the negation. Also, `$range` variable is redundant